overlord/configstate/configcore/sysctl.go

```diff
@@ -91,8 +92,14 @@ func handleSysctlConfiguration(tr config.ConfGetter, opts *fsOnlyContext) error
 		}
 	}
 
-	// write the new config
 	dir := filepath.Join(root, sysctlConfsDir)
+	if opts != nil && !osutil.IsDirectory(dir) {
```

Comment:
Technically this is the wrong way to do this. Just create the directory and handle `EEXIST`. Otherwise the code is racy. This is a special case, because MkdirAll already checks for EEXIST, but the IsDirectory check is redundant.